runtime/io.js

```diff
@@ -138,8 +138,34 @@ function caml_ml_open_descriptor_in (fd)  {
   var refill = null;
   if(fd == 0 && fs_node_supported()){
     var fs = require('fs');
+    var Buffer = require('buffer').Buffer;
     refill = function () {
-      return caml_string_of_jsstring(fs.readFileSync(0, 'utf8'))};
+      var buf = Buffer.alloc(256);
```

Comment:
Should we allocate a new Buffer everytime we refill ?

Reply:
@hhugo this is a security risk because the buffer will contain extra information from the previous read. Node core has long recommended fresh or clear buffers for inputs because they will zero out all data in the buffer. There was a concerted effort in the JS ecosystem to do this work.